src/components/com_tjcertificate/administrator/libraries/certificate.php

```diff
@@ -962,15 +963,54 @@ public static function canDownload($uniqueCertificateId)
 
 		if ($user->authorise('certificate.download.own', 'com_tjcertificate'))
 		{
-			$table = TJCERT::table("certificates");
-			$table->load(array('unique_certificate_id' => $uniqueCertificateId));
-
-			if ($user->get('id') == $table->user_id)
+			if ($user->get('id') == $this->user_id)
 			{
 				return true;
 			}
 		}
 
 		return false;
 	}
+
+	/**
+	 * Method to get linkedIn add to profile url.
+	 *
+	 * @return  STRING
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function getLinkedInProfileUrl()
+	{
+		$params   = ComponentHelper::getParams('com_tjcertificate');
+		$config   = Factory::getConfig();
+		$siteName = $config->get('sitename');
+
+		$issuedMonth = HTMLHelper::_('date', $this->issued_on, 'm');
+		$issuedYear  = HTMLHelper::_('date', $this->issued_on, 'Y');
+
+		$expirationDetails = null;
+
+		if ($this->expired_on != '0000-00-00 00:00:00')
+		{
+			$expirationMonth   = HTMLHelper::_('date', $this->expired_on, 'm');
+			$expirationYear    = HTMLHelper::_('date', $this->expired_on, 'Y');
+			$expirationDetails = '&expirationYear=' . $expirationYear . '&expirationMonth=' . $expirationMonth;
+		}
+
+		$orgParam = '&' . $params->get('organization_info') . '=' . $params->get('organization_id_name');
+
+		// Get client data
+		$dispatcher = JDispatcher::getInstance();
+		PluginHelper::importPlugin('content');
+		$result = $dispatcher->trigger('getCertificateClientData', array($this->client_id, $this->client));
+		$clientData = $result[0];
+
+		$certificateTitle = $clientData->title ? $clientData->title : $siteName . ' ' . Text::_('COM_TJCERTIFICATE_CERTIFICATE_DETAIL_VIEW_HEAD');
+		$certificateUrl = JUri::root() . substr(Route::_($this->getURL('', false)), strlen(Uri::base(true)) + 1);
```

Comment:
$urlOptions = array();
$urlOptions['absolute'] = true;
$certificateUrl = $this->getURL($urlOptions, false)

Reply:
done